lib/chat/ui/widgets/bubble_message.dart

```diff
@@ -0,0 +1,184 @@
+      textStyle: textStyle ?? this.textStyle,
+      textContainerBackgroundColor:
+          textContainerBackgroundColor ?? this.textContainerBackgroundColor,
+      rowCrossAxisAlignment:
+          rowCrossAxisAlignment ?? this.rowCrossAxisAlignment,
+      textContainerMargin: textContainerMargin ?? this.textContainerMargin,
+      outerContainerMargin: outerContainerMargin ?? this.outerContainerMargin,
+      textContainerPadding: textContainerPadding ?? this.textContainerPadding,
+      rowMainAxisAlignment: rowMainAxisAlignment ?? this.rowMainAxisAlignment,
+    );
+  }
+}
+
+class _DefaultMessageBubbleStyle extends MessageBubbleStyle {
+  final EdgeInsetsGeometry outerContainerMargin = const EdgeInsets.symmetric(
+    vertical: 10.0,
+    horizontal: 20.0,
+  );
+  final BorderRadiusGeometry textContainerBorderRadius =
+      const BorderRadius.all(Radius.circular(20.0));
+  final EdgeInsetsGeometry textContainerPadding = const EdgeInsets.all(16.0);
+  final EdgeInsetsGeometry textContainerMargin = const EdgeInsets.only(
+    left: 30.0,
+    right: 0.0,
+  );
+  final Color textContainerBackgroundColor = const Color(0xFF00CD9F);
+  final TextAlign textAlignment = TextAlign.start;
+  final TextStyle textStyle = const TextStyle(
+    color: const Color(0xFFFFFFFF),
+    fontSize: 15.0,
+  );
+  final CrossAxisAlignment rowCrossAxisAlignment = CrossAxisAlignment.end;
+  final MainAxisAlignment rowMainAxisAlignment = MainAxisAlignment.end;
+
+  const _DefaultMessageBubbleStyle({
+    EdgeInsetsGeometry outerContainerMargin,
+    BorderRadiusGeometry textContainerBorderRadius,
+    EdgeInsetsGeometry textContainerPadding,
+    EdgeInsetsGeometry textContainerMargin,
+    Color textContainerBackgroundColor,
+    TextAlign textAlignment,
+    TextStyle textStyle,
+    CrossAxisAlignment rowCrossAxisAlignment,
+    MainAxisAlignment rowMainAxisAlignment,
+  });
+}
+
+const MessageBubbleStyle _defaultStyle = const _DefaultMessageBubbleStyle();
+
+class MessageBubble extends StatelessWidget {
+  final MessageBubbleViewModel _viewModel;
+  final MessageBubbleStyle _style;
+  final Function _onTap;
+
+  MessageBubble({
+    @required MessageBubbleViewModel viewModel,
+    MessageBubbleStyle style,
+    Function onTap,
+  })  : this._viewModel = viewModel,
+        this._style = style ?? _defaultStyle,
+        this._onTap = onTap ?? (() => {});
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+        onTap: _onTap,
+        child: Container(
+          margin: _style.outerContainerMargin,
+          child: Row(
+            mainAxisSize: MainAxisSize.min,
+            crossAxisAlignment: _style.rowCrossAxisAlignment,
+            mainAxisAlignment: _style.rowMainAxisAlignment,
+            children: <Widget>[
+              _buildAvatar(),
+              _buildContainer(),
+            ].where(_notNull).toList(),
+          ),
+        ));
+  }
+
+  bool _notNull(Widget item) => item != null;
```

Comment:
move this to the top (below the finals)

Reply:
Moved. Do not really understand the purpose of doing so, is a function...